infer/lib/python/inferlib/capture/gradle.py

```diff
@@ -31,9 +30,136 @@ def gen_instance(*args):
 create_argparser = util.base_argparser(MODULE_DESCRIPTION, MODULE_NAME)
 
 
+def extract_filepath(parts):
+    pos = 0
```

Comment:
Please do. I guess the original order has been used to avoid use-cases like colliding "path to file.java" and "path/ to file.java" (substring of the first path is a directory for the second one). However, tests do not fail and if it works for you, I doubt we should care about minor use-case (if it's even possible, considering Javac requirements for the file names).